lib/main.dart

```diff
@@ -6,59 +6,61 @@ void main() {
   runApp(const ScreenerApp());
 }
 
-class ScreenerApp extends StatelessWidget {
+class ScreenerApp extends StatefulWidget {
   const ScreenerApp({Key? key}) : super(key: key);
 
   @override
-  Widget build(BuildContext context) {
-    const _appTitle = "Screener";
-    return const MaterialApp(
-        debugShowCheckedModeBanner: false,
-        title: _appTitle,
-        home: ScreenerHome(),
-    );
-  }
-}
-
-class ScreenerHome extends StatelessWidget {
-  const ScreenerHome({Key? key}) : super(key: key);
-
-  @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: PreferredSize(
-          preferredSize: const Size.fromHeight(0),
-          child: AppBar(
-            title: const SizedBox(
-              height: kToolbarHeight,
-            ),
-          )),
-      body: const ScreenerWebView(),
-    );
-  }
+  State<ScreenerApp> createState() => _ScreenerAppState();
 }
 
-class ScreenerWebView extends StatelessWidget {
-  const ScreenerWebView({Key? key}) : super(key: key);
+class _ScreenerAppState extends State<ScreenerApp> {
+  late WebViewController controller;
 
   @override
   Widget build(BuildContext context) {
     const _screenerHomeUrl = "https://www.screener.in";
     const _proxyUserAgent =
         "Mozilla/5.0 (Linux; Android 4.1.1; Galaxy Nexus Build/JRO03C) AppleWebKit/535.19 (KHTML, like Gecko) Chrome/18.0.1025.166 Mobile Safari/535.19";
-    return WebView(
-      initialUrl: _screenerHomeUrl,
-      javascriptMode: JavascriptMode.unrestricted,
-      userAgent: _proxyUserAgent,
-      navigationDelegate: (NavigationRequest request) {
-        if (request.url.startsWith(_screenerHomeUrl)) {
-          return NavigationDecision.navigate;
-        } else {
-          _launchURL(request.url);
-          return NavigationDecision.prevent;
-        }
-      },
-    );
+    return MaterialApp(
```

Comment:
Let's maintain smaller widgets as before?

Reply:
Hey @manujagobind  i tried doing this same thing with smaller widget tree, maybe I am not well versed with flutter but it took me 2-3 days and still I wasn't able to figure out, 
So I switched back it and got it done, and it was easier that way for me

Reply:
Ok, no worries. I'll give it a try on this branch.

Reply:
Hey @manujagobind don't worry about it, I'll do it all at the end,
It's easy thing to do, I've to just extract the widgets with 'Ctrl + .'
The thing is I find it hard to work that way, so let's keep it simple for now?

Reply:
I can help you understand the split.

Reply:
sure, but not like i don't get the split, I can read it and get what it does, but it's just hard working with it
for example, try adding the back navigation with the split

I can do the split with "Ctrl +."
Maybe do that towards the end?

Reply:
Widgets split is a standard idea. The longer the widgets, the higher chances that widget tree takes time to render, if there are many states across the unit widgets. 
Extracting the widgets into smaller ones, delegates state handling to the atomic widgets and not to the whole widget tree, hence optimised rendering as well, apart from just code readability.

Reply:
Hey @ashank96 can you push the changes?